tools/topology/topology2/development/tplg-targets.cmake

```diff
@@ -147,3 +133,33 @@ PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-tgl-nocodec-crossover.bin,EFX_CROSSOVE
 "development/cavs-nocodec-rtcaec\;sof-tgl-nocodec-rtcaec\;PLATFORM=tgl,\
 PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-tgl-nocodec-rtcaec.bin"
 )
+
+#
+# Append test topologies for multiple platforms and multiple formats
+# to TPLGS array.
+#
+
+set(platforms "TGL" "MTL")
+set(sampleformats "16" "24" "32")
+
+foreach(p ${platforms})
+	# Add components with all sample formats
+	foreach(sf ${sampleformats})
+		string(TOLOWER ${p} pl)
+		set(items
+			"sof-hda-generic\;sof-hda-benchmark-dcblock${sf}-${pl}\;PLATFORM=${p},HDA_CONFIG=benchmark,BENCH_CONFIG=dcblock${sf},BENCH_DCBLOCK_PARAMS=default"
+			"sof-hda-generic\;sof-hda-benchmark-drc${sf}-${pl}\;PLATFORM=${p},HDA_CONFIG=benchmark,BENCH_CONFIG=drc${sf},BENCH_DRC_PARAMS=enabled"
+			"sof-hda-generic\;sof-hda-benchmark-eqiir${sf}-${pl}\;PLATFORM=${p},HDA_CONFIG=benchmark,BENCH_CONFIG=eqiir${sf},BENCH_EQIIR_PARAMS=highpass_50hz_0db_48khz"
+			"sof-hda-generic\;sof-hda-benchmark-eqfir${sf}-${pl}\;PLATFORM=${p},HDA_CONFIG=benchmark,BENCH_CONFIG=eqfir${sf},BENCH_EQFIR_PARAMS=loudness"
+			"sof-hda-generic\;sof-hda-benchmark-gain${sf}-${pl}\;PLATFORM=${p},HDA_CONFIG=benchmark,BENCH_CONFIG=gain${sf}"
+			"sof-hda-generic\;sof-hda-benchmark-multiband_drc${sf}-${pl}\;PLATFORM=${p},HDA_CONFIG=benchmark,BENCH_CONFIG=multiband${sf},BENCH_MULTIBAND_DRC_PARAMS=default"
+		)
+		list(APPEND TPLGS "${items}")
```

Comment:
Are you sure this works as intended? I know CMake "pseudo-lists" are full of (bad) surprises but I would have expected $items to be unquoted here as in: `list(APPEND TPLGS ${items})`. Right now it looks like it's appending a single and very long string, not appending a list of configs (which is itself an embedded \\; list keywords, oh my...)

There's also a lot of repetition still and I think you can probably kill two birds with one stone. How about this:

```cmake
set(param_suffixes
			"BENCH_CONFIG=dcblock${sf},BENCH_DCBLOCK_PARAMS=default"
			"BENCH_CONFIG=drc${sf},BENCH_DRC_PARAMS=enabled"
    ....
)

foreach(suffix IN ${param_suffixes)
   list APPEND(TPLGS "sof-hda-generic\;sof-hda-benchmark-eqfir${sf}-${pl}\;PLATFORM=${p},HDA_CONFIG=benchmark\;${suffix}")
endforeach()

```

This
- finally gets rid of all the duplication and makes the variable bits much more visible and readable
- This APPENDs a single element at a time which mitigates the usual list&quoting headaches with CMake.

Reply:
> Are you sure this works as intended? I know CMake "pseudo-lists" are full of (bad) surprises but I would have expected $items to be unquoted here as in: `list(APPEND TPLGS ${items})`. Right now it looks like it's appending a single and very long string, not appending a list of configs (which is itself an embedded \; list keywords, oh my...)

I tried it first, of course, It didn't work without quotes. I have no idea why. But I'll try your suggestions, maybe things gets magically fixed.